app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationViewModel.kt

```diff
@@ -658,19 +754,36 @@ class ConversationViewModel(
 
     fun acceptMessageRequest() = viewModelScope.launch {
         val recipient = recipient ?: return@launch Log.w("Loki", "Recipient was null for accept message request action")
+        val currentState = _uiState.value.messageRequestState as? MessageRequestUiState.Visible
+            ?: return@launch Log.w("Loki", "Current state was not visible for accept message request action")
+
+        _uiState.update {
+            it.copy(messageRequestState = MessageRequestUiState.Pending(currentState))
+        }
+
         repository.acceptMessageRequest(threadId, recipient)
             .onSuccess {
                 _uiState.update {
-                    it.copy(isMessageRequestAccepted = true)
+                    it.copy(messageRequestState = MessageRequestUiState.Invisible)
                 }
             }
             .onFailure {
-                Log.w("", "Failed to accept message request: $it")
+                showMessage("Couldn't accept message request due to error: $it")
+
+                _uiState.update { state ->
+                    state.copy(messageRequestState = currentState)
+                }
             }
     }
 
-    fun declineMessageRequest() {
-        repository.declineMessageRequest(threadId)
+    fun declineMessageRequest() = viewModelScope.launch {
+        repository.declineMessageRequest(threadId, recipient!!)
+            .onSuccess {
+                _uiState.update { it.copy(shouldExit = true) }
+            }
+            .onFailure {
+                showMessage("Couldn't decline message request due to error: $it")
```

Comment:
We shouldn't hardcode a string as this won't be translated. We need to use a crowdIn equivalent